morango/models.py

```diff
@@ -437,6 +452,24 @@ def save(self, update_dirty_bit_to=True, *args, **kwargs):
             self._morango_dirty_bit = False
         super(SyncableModel, self).save(*args, **kwargs)
 
+    def delete(self, using=None, keep_parents=False, hard_delete=False, *args, **kwargs):
```

Comment:
A lot of this logic is taken from django's standard `delete` method. This adds the tracking of IDs of the related models to also hard delete.